server/src/test/java/org/apache/iotdb/db/integration/IoTDBQueryMemoryControlIT.java

```diff
@@ -0,0 +1,213 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.integration;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSetMetaData;
+import java.sql.Statement;
+import org.apache.iotdb.db.conf.IoTDBDescriptor;
+import org.apache.iotdb.db.utils.EnvironmentUtils;
+import org.apache.iotdb.jdbc.Config;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class IoTDBQueryMemoryControlIT {
+
+  private static final String[] sqls = new String[]{
+      "set storage group to root.ln",
+
+      "create timeseries root.ln.wf01.wt01 with datatype=BOOLEAN,encoding=PLAIN",
+      "create timeseries root.ln.wf01.wt02 with datatype=BOOLEAN,encoding=PLAIN",
+      "create timeseries root.ln.wf01.wt03 with datatype=BOOLEAN,encoding=PLAIN",
+      "create timeseries root.ln.wf01.wt04 with datatype=BOOLEAN,encoding=PLAIN",
+      "create timeseries root.ln.wf01.wt05 with datatype=BOOLEAN,encoding=PLAIN",
+
+      "create timeseries root.ln.wf02.wt01 with datatype=FLOAT,encoding=RLE",
+      "create timeseries root.ln.wf02.wt02 with datatype=FLOAT,encoding=RLE",
+      "create timeseries root.ln.wf02.wt03 with datatype=FLOAT,encoding=RLE",
+      "create timeseries root.ln.wf02.wt04 with datatype=FLOAT,encoding=RLE",
+      "create timeseries root.ln.wf02.wt05 with datatype=FLOAT,encoding=RLE",
+
+      "create timeseries root.ln.wf03.wt01 with datatype=TEXT,encoding=PLAIN",
+      "create timeseries root.ln.wf03.wt02 with datatype=TEXT,encoding=PLAIN",
+      "create timeseries root.ln.wf03.wt03 with datatype=TEXT,encoding=PLAIN",
+      "create timeseries root.ln.wf03.wt04 with datatype=TEXT,encoding=PLAIN",
+      "create timeseries root.ln.wf03.wt05 with datatype=TEXT,encoding=PLAIN",
+  };
+
+  @Before
+  public void setUp() throws Exception {
+    IoTDBDescriptor.getInstance().getConfig().setMaxQueryDeduplicatedPathNum(10);
+    EnvironmentUtils.envSetUp();
+    Class.forName(Config.JDBC_DRIVER_NAME);
+    createTimeSeries();
+  }
+
+  private static void createTimeSeries() {
+    try (Statement statement = DriverManager.getConnection(
+        Config.IOTDB_URL_PREFIX + "127.0.0.1:6667/", "root", "root").createStatement()) {
+      for (String sql : sqls) {
+        statement.execute(sql);
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
+      fail(e.getMessage());
+    }
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    EnvironmentUtils.cleanEnv();

```

Comment:
Don't forget to change the MaxQueryDeduplicatedPathNum back to the default value.